.github/workflows/ci.yml

```diff
@@ -719,7 +719,12 @@ jobs:
         run: |
           find . -type f -name \*-executable.jar -exec chmod 0777 {} \;
       - name: Enable impact analysis
-        if: steps.filter.outputs.product-tests == 'false' && !contains(github.event.pull_request.labels.*.name, 'tests:all') && !contains(github.event.pull_request.labels.*.name, 'tests:all-product')
+        if: |
+          steps.filter.outputs.product-tests == 'false' &&
+          github.event_name == 'pull_request' &&
+          github.event.pull_request.head.repo.full_name != github.repository &&
```

Comment:
Is it "do not run impact-analyis from PR coming from the origin"?
Why so?

Reply:
I think logic is ok. But please add a comment maybe.

Reply:
PTAL

Reply:
awesome!